### PR TITLE
Migrate frontend/inventory.go to use Definition_Rack

### PR DESCRIPTION
**Pull request by Jim3Things**
... and retire the now obsolete DbInventory.GetRack function

### CODE REVIEW

File: simulation/internal/services/frontend/inventory.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ func handlerRacksList(w http.ResponseWriter, r *http.Request) {
 	memoData := dbInventory.GetMemoData()
 	if memoData == nil {
 		postHTTPError(ctx, w, errors.ErrInventoryNotAvailable)
+		return
```
Comment (MikeCJohnsonUk) [inline simulation/internal/services/frontend/inventory.go:73]: I presume this was the cause of the double-error update you were seeing?

File: simulation/internal/services/frontend/inventory.go [hunk 2]
```diff
@@ -219,17 +236,50 @@ func handlerBladeRead(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	for i, blade := range rd.Blades {
+		b := &pb.BladeCapacity{
+			Cores:                  blade.Capacity.Cores,
+			MemoryInMb:             blade.Capacity.MemoryInMb,
+			DiskInGb:               blade.Capacity.DiskInGb,
+			NetworkBandwidthInMbps: blade.Capacity.NetworkBandwidthInMbps,
+			Arch:                   blade.Capacity.Arch,
+			Accelerators:           nil,
```
Comment (MikeCJohnsonUk) [inline simulation/internal/services/frontend/inventory.go:274]: You should be able to replace this piece with a

b := blade.GetCapacity()

which should also clone the accelerators array.


Actually, this should all be reducible to 

rack.Blades[i] = blade.GetCapacity()

Comment (MikeCJohnsonUk) [reply simulation/internal/services/frontend/inventory.go:274]: For ref see lines 353 to 355 in the old GetRack()

Comment (Jim3Things) [reply simulation/internal/services/frontend/inventory.go]: done

File: simulation/internal/services/frontend/inventory.go (hunk 1 above)
Comment (Jim3Things) [reply simulation/internal/services/frontend/inventory.go:73]: I think maybe...

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**